tests/conftest.py

```diff
@@ -1,7 +1,127 @@
+from numpy.typing import NDArray
+
+
+def _make_stack(
+    offset: int | None = None,
+    mask: bool = False,
+) -> NDArray[np.float64]:
+    """Create a 50x50x50 zeros stack with foreground."""
+
+    shape = [50, 50, 50]
+    obj_size = 20
+    mask_extra = 5
+    value = 1 if mask else 0.5
+
+    stack = np.zeros(shape, dtype=np.float64)
+    foreground_size = obj_size + (mask_extra if mask else 0)
+
+    start = [(s - foreground_size) // 2 for s in shape]
+    if offset:
+        start = [s - offset for s in start]
+    end = [s + foreground_size for s in start]
+
+    stack[start[0] : end[0], start[1] : end[1], start[2] : end[2]] = value
+
+    return stack
+
+
+def _create_test_images(
+    path: Path,
+    test_data: list[dict[str, Any]],
+    image_type: str = "nifti",
+) -> list[dict[str, Any]]:
+    """Save test images and add their paths to the returned test data dicts."""
+
+    file_ext = ".nii.gz" if image_type == "nifti" else ".tif"
+
+    for data in test_data:
+        subject_dir = path / data["subject_id"]
+        subject_dir.mkdir(parents=True, exist_ok=True)
+
+        data["filepath"] = subject_dir / f"{data['subject_id']}{file_ext}"
+        stacks = {"image": (data["image"], data["filepath"])}
+
+        if data["mask"] is not None:
+            data["mask_filepath"] = (
+                subject_dir / f"{data['subject_id']}_mask{file_ext}"
+            )
+            stacks["mask"] = (data["mask"], data["mask_filepath"])
+
+        for _, (stack, stack_path) in stacks.items():
+            if image_type == "nifti":
+                save_as_asr_nii(
+                    stack,
+                    [v * 0.001 for v in data["voxel_size"]],
+                    stack_path,
+                )
+            else:
+                save_any(stack, stack_path)
+
+    return test_data
+
+
+def _create_test_csv(
+    path: Path, test_data: list[dict[str, Any]], csv_name: str
+) -> Path:
+    """Create CSV file and returns its path."""
+
+    data_dict = test_data.copy()
+    for data in data_dict:
+        data["resolution_0"] = data["voxel_size"][0]
+        data["resolution_1"] = data["voxel_size"][1]
+        data["resolution_2"] = data["voxel_size"][2]
+        for k in ("voxel_size", "image", "mask"):
+            data.pop(k)
+
+    csv_path = path / f"{csv_name}.csv"
+    pd.DataFrame(data=data_dict).to_csv(csv_path, index=False)
+    return csv_path
+
+
+def _create_test_yaml(path: Path) -> Path:
+    """Create YAML config file and returns its path."""
+    yaml_dict = {
+        "output_dir": str(path.resolve()),
+        "mask": {
+            "gaussian_sigma": 3,
+            "threshold_method": "triangle",
+            "closing_size": 5,
+            "erode_size": 0,
+        },
+        "pad_pixels": 5,
+    }
+
+    config_path = path / "config.yaml"
+    with open(config_path, "w") as outfile:
+        yaml.dump(yaml_dict, outfile, default_flow_style=False)
+
+    return config_path
+
+
+def _write_test_data(
```

Comment:
```suggestion
@pytest.fixture
def write_test_data(
```
as implied by the other comments, but will also need some like refactoring so it returns an inner function instead of the csv path.